docs/index.rst

```diff
@@ -268,6 +268,35 @@ then the *hookimpl* should be marked with the ``"optionalhook"`` option:
 
         return config
 
+.. _specname:
+
+Hook implementation naming
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+During plugin :ref:`registration <registration>`, hook implementations are
+matched to a hook :ref:`specifications <specs>` in the *host* program with the
```

Comment:
matched to a hook :ref:`specifications <specs>` **if provided** in the *host*..

Reply:
to make the sentence read a bit better, I broke into two sentences, restated that there is no string 1-1 requirement, and linked to "enforcing spec validation".